ios/Gekidou/Sources/Gekidou/Storage/Database+Posts.swift

```diff
@@ -143,15 +164,21 @@ extension Database {
         return (0, 0)
     }
     
-    public func handlePostData(_ db: Connection, _ postData: PostData, _ channelId: String, _ usedSince: Bool = false) throws {
+    public func handlePostData(_ db: Connection, _ postData: PostData, _ channelId: String, _ usedSince: Bool = false, _ receivingThreads: Bool = false) throws {
         let sortedChainedPosts = chainAndSortPosts(postData)
         try insertOrUpdatePosts(db, sortedChainedPosts, channelId)
         let earliest = sortedChainedPosts.first!.create_at
         let latest = sortedChainedPosts.last!.create_at
-        try handlePostsInChannel(db, channelId, earliest, latest, usedSince)
+        if (!receivingThreads) {
+            try handlePostsInChannel(db, channelId, earliest, latest, usedSince)
+        }
         try handlePostsInThread(db, postData.posts)
     }
     
+    public func handleThreads(_ db: Connection, _ threads: [Post]) throws {
+        try insertThreads(db, threads)
```

Comment:
on Android you are either inserting or updating, why is that not the case for iOS?

Reply:
If we replace it .. we might lose the existing data as we don't get all the thread data. I couldn't find it in the existing code but some sort of "MERGING" might also work.

In Android, I was able to get the existing records and was using existing values as a fallback.

Reply:
but can't you get the records as is done on Android ?